clients/src/main/java/org/apache/kafka/common/utils/Bytes.java

```diff
@@ -139,7 +140,7 @@ private static String toString(final byte[] b, int off, int len) {
      */
     public final static Comparator<byte[]> BYTES_LEXICO_COMPARATOR = new LexicographicByteArrayComparator();
 
-    private interface ByteArrayComparator extends Comparator<byte[]> {
+    private interface ByteArrayComparator extends Comparator<byte[]>, Serializable {
```

Comment:
We normally don't make things `Serializable` unless there's a good reason to do it. Is there a good reason?